src/napari_live_recording/control/__init__.py

```diff
@@ -43,32 +49,36 @@ class MainController(QObject):
     recordFinished = Signal()
     newTimePoint = Signal(int)
     newMaxTimePoint = Signal(int)
+    cameraDeleted = Signal(bool)
 
     def __init__(self) -> None:
         """Main Controller class. Stores all camera objects to access live and stack recordings."""
         super().__init__()
         self.deviceControllers: Dict[str, LocalController] = {}
         self.deviceLiveBuffer: Dict[str, np.ndarray] = {}
+        self.deviceBuffers: Dict[str, Framebuffer] = {}
+        self.recordingBuffers: Dict[str, Framebuffer] = {}
+        self.processingBuffers: Dict[str, Framebuffer] = {}
+        self.processedBuffers: Dict[str, Framebuffer] = {}
+        self.settings = Settings()
+        self.filtersDict = filtersDict
+        self.stackSize = 1000
+        self.idx = 0
         self.liveWorker = None
-        self.__isLive = False
+        self.__isAcquiring = {}
         self.recordLoopEnabled = False
+        self.processingRunning: Dict[str, bool] = {}
+        self.recordingRunning: Dict[str, bool] = {}
         self.recordSignalCounter = SignalCounter()
-        self.recordSignalCounter.maxCountReached.connect(lambda: self.recordFinished.emit())
-    
-    @property
-    def isLive(self) -> bool:
-        return self.__isLive
+        self.recordSignalCounter.maxCountReached.connect(
+            lambda: self.recordFinished.emit()
+        )
+        self.recordFinished.connect(self.resetRecordingCounter)
+        self.bufferWorkers: Dict[str, FunctionWorker] = {}
 
-    @contextmanager
```

Comment:
Did you remove this context manager somehow?

Reply:
What exactly is the purpose of this ?